csrank/core/fate_network.py

```diff
@@ -176,7 +178,7 @@ def __init__(self, n_hidden_set_layers=1, n_hidden_set_units=1, **kwargs):
             kernel_initializer=self.kernel_initializer,
             kernel_regularizer=self.kernel_regularizer_,
         )
-        self.is_variadic = True
+        self.is_variadic = is_variadic
```

Comment:
@kiudee @prithagupta what is the intention behind this attribute? It was hardcoded to `True` before and not documented, so I'm having a hard time understanding its function.

Reply:
There is some code for the non-variadic case though (which could never be reached). Should I just remove that?

Reply:
While trying to remove it I noticed that it *could* be reached after all: If you pass something that is not a `dict` as `X` to `fit`, it would set `is_variadic` to `False`.

Reply:
Accordingly I removed it from `__init__` now and gave it an underscore postfix. Its set in `fit` and part of the variables that are "learned" from the data.